src/HIE/Bios/Cradle.hs

```diff
@@ -783,31 +810,42 @@ cabalWorkDir wdir =
       findFileUpwards (== "cabal.project") wdir
   <|> findFileUpwards (\fp -> takeExtension fp == ".cabal") wdir
 
+
 ------------------------------------------------------------------------
 
--- | Explicit data-type for stack.yaml configuration location.
+-- | Explicit data-type for project configuration location.
 -- It is basically a 'Maybe' type, but helps to document the API
 -- and helps to avoid incorrect usage.
-data StackYaml
-  = NoExplicitYaml
-  | ExplicitYaml FilePath
+data CradleProjectConfig
+  = NoExplicitConfig
+  | ExplicitConfig FilePath
+
+instance Semigroup CradleProjectConfig where
+  NoExplicitConfig <> s = s
+  s@(ExplicitConfig _) <> _ = s
```

Comment:
suspicious. Should we just use `First CradleProjectConfig`?

Reply:
These instances aren't used anyway, maybe we should just remove them